agenthub/codeact_agent/codeact_agent.py

```diff
@@ -153,6 +153,8 @@ def get_observation_message(self, obs: Observation) -> Message | None:
             text = truncate_content(text, max_message_chars)
             return Message(role='user', content=[TextContent(text=text)])
         elif isinstance(obs, AgentDelegateObservation):
+            if isinstance(obs.outputs, dict) and 'content' in obs.outputs:
+                obs.outputs = obs.outputs['content']
             text = obs_prefix + truncate_content(str(obs.outputs), max_message_chars)
```

Comment:
```suggestion
            text = obs_prefix + truncate_content(obs.outputs['content'] if 'content' in obs.outputs else ''), max_message_chars)
```

Because the `AgentController` will always throw a dict, maybe we don't need to check again?
https://github.com/All-Hands-AI/OpenHands/blob/45066f19dcec21b718e1d5dfb0586ae1c3091a90/openhands/controller/agent_controller.py#L470-L488